### PR TITLE
Benchmarks improvements

### PR DESCRIPTION
**Pull request by arsh**
## Description of change

- Changing the remaining of runners from `high-performance` to `nvme-high-performance`
- Updating benchmark documentation
- Making caching benchmarks use the same jobs as read, eliminating the duplication
- Removing start delay of `fio` jobs

Relevant issues: N/A

## Does this change impact existing behavior?

No

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: doc/BENCHMARKING.md [hunk 1]
```diff
@@ -38,6 +38,10 @@ You can use the following steps.
         export S3_BUCKET_TEST_PREFIX=prefix_path/
         export S3_BUCKET_BENCH_FILE=bench_file_name
         export S3_BUCKET_SMALL_BENCH_FILE=small_bench_file_name
+        # to filter by job name; e.g. to only run small jobs
+        export S3_JOB_NAME_FILTER=small
+        # to skip setting up local EC2 storage for caching benchmarks; the value is not important
+        export S3_SKIP_STORAGE_SETUP=
 
 3. Create the bench files manually in your S3 bucket. You can do that by mounting the bucket on your machine using Mountpoint. Then, running fio jobs against your mount directory to let fio create the files for you.
```
Comment (sauraank) [inline doc/BENCHMARKING.md:46]: Now we dont need this step.

File: doc/BENCHMARKING.md [hunk 2]
```diff
@@ -18,9 +18,9 @@ In general, we run each IO operation for 30 seconds against a 100 GiB file. But
...
+Our CI runs the benchmark automatically for any new commits to the main branch or specific pull requests that we have reviewed and tagged with **performance** label. Every benchmark from the CI workflow will be running on `m5dn.24xlarge` EC2 instances (100 Gbps network speed) with Ubuntu 22.04 in us-east-1 against a bucket in us-east-1.
 
-We keep the records of benchmarking results in `gh-pages` branch and the performance charts are available for viewing as [throughput chart](https://awslabs.github.io/mountpoint-s3/dev/bench/) and [latency chart](https://awslabs.github.io/mountpoint-s3/dev/latency_bench/).
+We keep the records of benchmarking results in `gh-pages` branch and the performance charts are available for viewing as [throughput chart](https://awslabs.github.io/mountpoint-s3/dev/bench/), [throughput (with cache) chart](https://awslabs.github.io/mountpoint-s3/dev/cache_bench/) and [latency chart](https://awslabs.github.io/mountpoint-s3/dev/latency_bench/).
```
Comment (monthonk) [inline doc/BENCHMARKING.md:23]: We now have many pages showing the benchmark results from various configs (S3 standard, S3 Express, S3 with cache), we might want to list them all here.

File: doc/BENCHMARKING.md [hunk 3]
```diff
@@ -38,6 +38,10 @@ You can use the following steps.
         export S3_BUCKET_TEST_PREFIX=prefix_path/
         export S3_BUCKET_BENCH_FILE=bench_file_name
         export S3_BUCKET_SMALL_BENCH_FILE=small_bench_file_name
+        # to filter by job name; e.g. to only run small jobs
+        export S3_JOB_NAME_FILTER=small
+        # to skip setting up local EC2 storage for caching benchmarks; the value is not important
+        export S3_SKIP_STORAGE_SETUP=
```
Comment (monthonk) [inline doc/BENCHMARKING.md:44]: I wonder if we should skip it by default just to make manual run a bit easier.

Comment (arsh) [reply doc/BENCHMARKING.md:44]: I'm optimizing for what runs more often which is the CI.

File: doc/BENCHMARKING.md (hunk 2 above)
Comment (arsh) [reply doc/BENCHMARKING.md:23]: I know Ankit is looking into combining the multiple results into the same graph. I'll ask him if we want to wait for that.

File: doc/BENCHMARKING.md (hunk 1 above)
Comment (arsh) [reply doc/BENCHMARKING.md:46]: Sure, I can remove the details on creating files.

File: doc/BENCHMARKING.md (hunk 3 above)
Comment (monthonk) [reply doc/BENCHMARKING.md:44]: But for the CI we only need 1 change to configure it correctly while we will have to configure it every time on manual runs.

Comment (arsh) [reply doc/BENCHMARKING.md:44]: Alright. I'll change it.

**Review by monthonk — APPROVED** [review]
LGTM!

**Pull request merged**